rootfs/scheduler/__init__.py

```diff
@@ -404,36 +456,36 @@ def run(self, name, image, entrypoint, command):
         resp = self.session.post(url, json=js_template)
         if unhealthy(resp.status_code):
             error(resp, 'create Pod in Namespace "{}"', appname)
-
-        while(1):
-            parsed_json = {}
-            status = 404
-            reason = ''
-            data = ''
-            for _ in range(5):
+        parsed_json = {}
+        status = 404
+        reason = ''
+        data = ''
+        duration = 30
+        iteration = 1
+        while(iteration < duration and duration < 90):
+            try:
                 status, reason, data = self._get_pod(name, appname)
-                if unhealthy(status):
-                    time.sleep(1)
-                    continue
-
                 parsed_json = json.loads(data)
+                if parsed_json['status']['phase'] == 'Succeeded':
+                    status, data, reason = self._pod_log(name, appname)
+                    self._delete_pod(name, appname)
+                    return 0, data
+                if parsed_json['status']['phase'] == 'Running':
+                    if iteration > 28:
+                        duration = duration + 1
+            except:
                 break
-
-            if unhealthy(status):
-                error(resp, 'create Pod in Namespace "{}"', appname)
-
-            if parsed_json['status']['phase'] == 'Succeeded':
-                status, data, reason = self._pod_log(name, appname)
-                self._delete_pod(name, appname)
-                return 0, data
-            elif parsed_json['status']['phase'] == 'Failed':
-                pod_state = parsed_json['status']['containerStatuses'][0]['state']
-                err_code = pod_state['terminated']['exitCode']
-                self._delete_pod(name, appname)
-                return err_code, data
-
+            iteration = iteration + 1
             time.sleep(1)
 
+        if duration == 90 or iteration >= duration:
```

Comment:
> If the user drops the connection do we lose total track of this job beyond it already being in k8s running?

Yes, we lose complete track of the job and it will continue running, but no cleanup code will occur. This is the same as closing the connection halfway through a `config:set` operation as well.